### ISSUE TITLE
Add variable type

### ISSUE THREAD

**Issue by gemal** [cakephp/debug_kit#771]
to be able to see if a variable is a int or string since it's impossible to see if a variable is "8". is that a string 8 or 8 int
today:
![image](https://user-images.githubusercontent.com/69105/89403707-079b6480-d719-11ea-886a-5c3aac8c3eec.png)

perhaps something like this:
![image](https://user-images.githubusercontent.com/69105/89403683-fe11fc80-d718-11ea-9592-36e6d0a51f88.png)

would that be something anybody would be interested in?

**Comment by markstory** [comment]
I think this is already available in the 4.x version of debugkit. Were you hoping to see this change in an earlier release?

**Comment by dereuromark** [comment]
I checked and didnt see it either, though.

**Comment by markstory** [comment]
@dereuromark hrm. Maybe I only thought about doing it and then forgot to actually do it.

**Comment by dereuromark** [comment]
![Screenshot from 2020-08-09 15-34-23](https://user-images.githubusercontent.com/39854/89733451-d5aa3b00-da55-11ea-855c-f78586db7f2b.png)

non-scalar ones seem to have that, and NULL
but not string/int etc, which are hard to tell the type of then.

**Comment by markstory** [comment]
Would using the new debug output from 4.1 work to help solve this? I think it could make output more consistent with the rest of CakePHP, fix this problem and give us less code to maintain in the long run.  This is the change that I had thought I did but didn't actually do.

**Comment by dereuromark** [comment]
PR opened for it.

**Comment by dereuromark** [comment]
@markstory Using it now I discovered a few follow ups

The root level should probably stay the former list of expandable items
Right now the whole huge array is dumped at once
```
[
    '_authUser' => [
],
    'queuedJobs' => [
],
    'queuedJob' => [
...
```
which can be thousands and millions of rows directly

The other issue is the tick box "Sort variables by name" above that now also become not functional due to this.

**Comment by markstory** [comment]
@dereuromark I think the collapsing is fixed in 4.1.3 I can verify that though. I'll check out the sort variables feature as that was done before the HTML was generated before.
